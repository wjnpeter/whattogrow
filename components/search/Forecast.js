import React, { useState } from 'react'
import _ from 'lodash'
import { IconButton, Popover, Typography, ButtonBase, Box, Grid, Avatar, Divider } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import LoadingCircular from '../LoadingCircular'

const convert = (temp, isFaren) => {
  const ret = isFaren ? temp * 9 / 5 + 32 : temp
  return Math.round(ret)
}

const dailyWeather = (data, isFaren) => {
  const max = Math.round(convert(data.temperatureHigh, isFaren))
  const min = Math.round(convert(data.temperatureLow, isFaren))
  return <>
    <Avatar variant='square' src={iconSrc(data.icon)} />
    <Typography variant="body1">
      {max + ' | ' + min}
    </Typography>
    <Typography
      display='block'
      variant='body1'
      paragraph
      style={{ width: '100%' }}>
      {data.summary}
    </Typography>
  </>
}

const iconSrc = (i) => '/images/icons/weather/' + i + '.png'

const useStyles = makeStyles((theme) => ({
  avatar: {
    width: theme.spacing(8),
    height: theme.spacing(8),
  },
  popover: {
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.primary.contrastText,
    width: 250,
    padding: theme.spacing(2, 4)
  },
  divider: {
    margin: theme.spacing(4, 6, 6)
  },
  unitSwitch: {
    position: 'absolute',
    top: theme.spacing(4),
    right: theme.spacing(4),
  },
  dimText: {
    color: theme.palette.text.disabled
  }
}));

export default function Forecast({ ds }) {

  const classes = useStyles();
  const [anchorEl, setAnchorEl] = useState(null);
  const [isFaren, setFaren] = useState(false)

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const open = Boolean(anchorEl);
  const id = open ? 'popover' : undefined;

  if (_.isNil(ds)) {
    return <LoadingCircular />
  }

  const { icon: curIcon, apparentTemperature: cutTemp } = ds.currently
  const { data: forecast } = ds.daily

  const unit = isFaren ? '째F' : '째C'
  return <>
    <IconButton onClick={handleClick} color='inherit'>
      <Avatar className={classes.avatar} variant='square' src={iconSrc(curIcon)} />
      <Typography variant="body1">{convert(cutTemp, isFaren) + unit}</Typography>
    </IconButton>

    <Popover
      id={id}
      open={open}
      anchorEl={anchorEl}
      onClose={handleClose}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'center',
      }}
      PaperProps={{ className: classes.popover }}
    >
      <Grid container direction='column' spacing={2}>
        <Grid item>
          <Typography variant='h6'>TODAY</Typography>
        </Grid>
        <Grid item container justify='space-between'>
          {dailyWeather(forecast[0], isFaren)}
        </Grid>
      </Grid>
      <Divider className={classes.divider} />

      <Grid container direction='column' spacing={2}>
        <Grid item>
          <Typography variant='h6'>TOMORROW</Typography>
        </Grid>
        <Grid item container justify='space-between'>
          {dailyWeather(forecast[1], isFaren)}
        </Grid>
      </Grid>

      <ButtonBase className={classes.unitSwitch} onClick={() => setFaren(!isFaren)}>
        <Typography variant="caption" children='째C'
          className={isFaren ? classes.dimText : ''}>
        </Typography>
        {'/'}
        <Typography variant="caption" children='째F'
          className={isFaren ? '' : classes.dimText} >
        </Typography>
      </ButtonBase>
    </Popover>
  </>
}