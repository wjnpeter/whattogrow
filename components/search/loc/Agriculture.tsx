
import useSWR from 'swr'
import axios from 'axios'
import { useContext } from 'react'
import { Container } from '@mui/material';

import AgriCard from './AgriCard'
import AppContext from '../../../lib/contexts'
import { fetchers } from '../../../lib/apiFetchers'

export default function Agriculture() {
  const appContext = useContext(AppContext)

  const token = appContext.token
  const geo = appContext.geo

  // Here's an assumption: 
  // The station from bomstn is the same as the 'list' from bomagri

  const agriData = (res: any) => res && res.data && res.data.agri

  const { data: stationTerrTemp } = useSWR(
    [geo, (124).toString(), token], 
    ([geo, product, token]) => fetchers.bomstn(geo, product, token)
  )
  const { data: agriTerrTemp } = useSWR(
    () => [geo, stationTerrTemp.data.station.id, token], 
    ([geo, station, token]) => fetchers.bomagri(geo, station, token)
  )
  const terrTemp = agriData(agriTerrTemp) ? agriData(agriTerrTemp).terrTemp : null

  const { data: stationSunshine } = useSWR(
    [geo, (133).toString(), token], 
    ([geo, product, token]) => fetchers.bomstn(geo, product, token)
  )
  const { data: agriSunshine } = useSWR(
    () => [geo, stationSunshine.data.station.id, token], 
    ([geo, station, token]) => fetchers.bomagri(geo, station, token)
  )
  const sunshine = agriData(agriSunshine) ? agriData(agriSunshine).sunshine : null

  const { data: stationEvaporation } = useSWR(
    [geo, (125).toString(), token], 
    ([geo, product, token]) => fetchers.bomstn(geo, product, token)
  )
  const { data: agriEvaporation } = useSWR(
    () => [geo, stationEvaporation.data.station.id, token], 
    ([geo, station, token]) => fetchers.bomagri(geo, station, token)
  )
  const evaporation = agriData(agriEvaporation) ? agriData(agriEvaporation).evaporation : null

  const precip = agriData(agriTerrTemp) ? agriData(agriTerrTemp).precip : null

  let cards = []
  if (terrTemp) {
    cards.push(
      <AgriCard
        key='terrTemp'
        icon='/images/icons/terrtmp.svg'
        number={terrTemp}
        label='Overnight Temperature'
        unit='Â°C'
      />
    )
  }

  if (precip) {
    cards.push(
      <AgriCard
        key='precip'
        icon='/images/icons/precip.svg'
        number={precip}
        label='Rainfall'
        unit='mm'
      />
    )
  }

  if (sunshine) {
    cards.push(
      <AgriCard
        key='sunshine'
        icon='/images/icons/sunshine.svg'
        number={sunshine}
        label='Sunshine Hours'
        unit='hours'
      />
    )
  }

  if (evaporation) {
    cards.push(
      <AgriCard
        key='Evaporation'
        icon='/images/icons/evaporation.svg'
        number={evaporation}
        unit='mm'
        label='Evaporation'
      />
    )
  }

  return <Container>
    {cards}
  </Container>
}
