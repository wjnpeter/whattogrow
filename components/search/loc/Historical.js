import useSWR from 'swr'

import { useContext, useState } from 'react'
import { makeStyles } from '@mui/styles';
import { Collapse, Typography, Grid } from '@mui/material';

import VLine from './VLine'
import Number from './Number'
import LocAvatar from './LocAvatar'
import HistoricalTriangle from './HistoricalTriangle'
import LoadingCircular from '../../LoadingCircular'
import { monthNames } from '../../../lib/utils'
import AppContext from '../../../lib/contexts'
import { fetchers } from '../../../lib/apiFetchers'

const useStyles = makeStyles((theme) => ({
  cardContent: {
    cursor: 'pointer',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    textAlign: 'center'
  },
  expand: {
    transform: 'rotate(0deg)',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  yrDetail: {
    backgroundColor: theme.palette.primary.light,
    padding: theme.spacing(3, 2)
  },
  section: {
    flexShrink: 0
  },
  sectionTitle: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(2),
  }
}))

const unit = product => product == 139 ? 'mm' : 'Â°C'

const cardType = {
  meanrain: 0,
  ytdrain: 1,
  maxtemp: 2,
  mintemp: 3
}

const content = (uiAbout, meanData) => {
  let number = ''
  let label = ''
  let sublabel = ''

  const data = meanData.data
  const stats = data.stats

  switch (uiAbout) {
    case cardType.meanrain:
      number = stats.annual
      label = 'Last ' + (stats.end - stats.start) + ' Years'
      sublabel = stats.start + ' ~ ' + stats.end
      break
    case cardType.ytdrain:
      const ytd = data.ytd
      number = ytd.annual
      label = 'Year To Day'
      const em = ytd.months.findIndex(m => m === 0)
      sublabel = 'JAN ~ ' + monthNames[em]
      break
    case cardType.maxtemp:
      number = stats.annual
      label = 'Mean Maximum'
      sublabel = 'Last ' + (stats.end - stats.start) + ' Years'
      break
    case cardType.mintemp:
      number = stats.annual
      label = 'Mean Minimum'
      sublabel = 'Last ' + (stats.end - stats.start) + ' Years'
      break
  }

  return <>
    <Number number={number} unit={unit(data.product)} />
    <Typography variant="body1" children={label} />
    <Typography variant="body1" children={sublabel} />
  </>
}

const yrDetail = (uiAbout, meanData, classes) => {
  const data = meanData.data

  const isYTD = uiAbout === cardType.ytdrain
  const months = isYTD ? data.ytd.months : data.stats.months

  return <>
    <Grid container spacing={1} className={classes.yrDetail}>
      {
        months.map((m, i) => {
          if (isYTD && m === 0) return

          return <Grid key={monthNames[i]} item xs={4}>
            <Typography
              variant="body1"
              children={monthNames[i] + ' ' + m + unit(data.about)}
            />
          </Grid>
        })
      }
    </Grid>
  </>
}

export default function Historical(props) {
  const classes = useStyles();
  const [expandedRain, setexpandedRain] = useState(null);
  const [expandedTemp, setexpandedTemp] = useState(null);
  const appContext = useContext(AppContext)

  const geo = appContext.geo
  const token = appContext.token

  const { data: rainStation } = useSWR(
    [geo, (139).toString(), token], 
    ([geo, product, token]) => fetchers.bomstn(geo, product, token)
  )
  const { data: meanRain } = useSWR(
    () => [rainStation.data.station.id, 139, token], 
    ([station, product, token]) => fetchers.bomstat(station, product, token)
  )

  const { data: tempStation } = useSWR(
    [geo, (122).toString(), token], 
    ([geo, product, token]) => fetchers.bomstn(geo, product, token)
  )
  // FIXME: 36 and 38 are not available
  const { data: meanMaxTemp } = useSWR(
    () => [tempStation.data.station.id, 36, token], 
    ([station, product, token]) => fetchers.bomstat(station, product, token)
  )
  const { data: meanMinTemp } = useSWR(
    () => [tempStation.data.station.id, 38, token], 
    ([station, product, token]) => fetchers.bomstat(station, product, token)
  )

  return (
    <>
      {meanRain === undefined ? <LoadingCircular /> :
        <section className={classes.section}>
          <Typography variant="body1" children={'Rainfall'} className={classes.sectionTitle} />

          <Grid container justify='center' alignItems='center' spacing={5}>
            <Grid item
              onClick={e => setexpandedRain(expandedRain === cardType.meanrain ? null : cardType.meanrain)}
              className={classes.cardContent} >
              <LocAvatar src="/images/icons/rainfall2.svg" />
              {content(cardType.meanrain, meanRain)}

              <HistoricalTriangle show={expandedRain === cardType.meanrain} />
            </Grid>

            <Grid item>
              <VLine />
            </Grid>

            <Grid item
              onClick={e => setexpandedRain(expandedRain === cardType.ytdrain ? null : cardType.ytdrain)}
              className={classes.cardContent} >
              <LocAvatar src="/images/icons/rainfall1.svg" />
              {content(cardType.ytdrain, meanRain)}

              <HistoricalTriangle show={expandedRain === cardType.ytdrain} />
            </Grid>
          </Grid>

          <Collapse in={expandedRain !== null} timeout="auto">
            {yrDetail(expandedRain, meanRain, classes)}
          </Collapse>
        </section>}

      {!meanMinTemp || !meanMaxTemp ? <LoadingCircular /> : <>
        <section className={classes.section}>
          <Typography variant="body1" children={'Temperature'} className={classes.sectionTitle}/>
          <Grid container justify='center' alignItems='center' spacing={5}>
            <Grid item
              onClick={e => setexpandedTemp(expandedTemp === cardType.maxtemp ? null : cardType.maxtemp)}
              className={classes.cardContent} >
              <LocAvatar src="/images/icons/maxtemp.svg" />
              {content(cardType.maxtemp, meanMaxTemp)}

              <HistoricalTriangle show={expandedTemp === cardType.maxtemp} />
            </Grid>

            <Grid item>
              <VLine />
            </Grid>

            <Grid item
              onClick={e => setexpandedTemp(expandedTemp === cardType.mintemp ? null : cardType.mintemp)}
              className={classes.cardContent} >
              <LocAvatar src="/images/icons/mintemp.svg" />
              {content(cardType.mintemp, meanMinTemp)}

              <HistoricalTriangle show={expandedTemp === cardType.mintemp} />
            </Grid>
          </Grid>

          <Collapse in={expandedTemp !== null} timeout="auto">
            {
              yrDetail(expandedTemp, expandedTemp === cardType.maxtemp ? meanMaxTemp : meanMinTemp, classes)
            }
          </Collapse>
        </section>
      </>}
    </>
  );
}