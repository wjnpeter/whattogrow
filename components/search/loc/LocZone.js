import useSWR from 'swr'
import axios from 'axios'

import { useContext } from 'react'
import Typography from '@material-ui/core/Typography';

import AppContext from '../../../lib/contexts'
import { fetchers } from '../../../lib/apiFetchers'



export default function LocZone() {
  const appContext = useContext(AppContext)

  const { data: koppen } = useSWR([appContext.geo, 'koppenmajor', appContext.token], fetchers.zone)

  const addr = appContext.addr
  if (!addr) return <></>

  const locality = addr.locality

  const state = addr && addr.state ? addr.state.short_name : ''
  const postcode = addr && addr.postcode ? addr.postcode.short_name : ''
  const tempZone = appContext.tempZone

  let zoneInfo = tempZone.descript
  if (koppen) {
    zoneInfo += '. ' + koppen.descript[0]
  }

  return <>
    <Typography variant="h6">
      {locality.short_name}
    </Typography>

    <Typography variant="body1" >{state + ' ' + postcode}</Typography>
    <Typography variant="body1">{zoneInfo}</Typography>


  </>
}