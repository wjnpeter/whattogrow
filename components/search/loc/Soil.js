import clsx from 'clsx';
import useSWR from 'swr'
import axios from 'axios'
import { useContext } from 'react'

import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';

import Number from './Number'
import {soilDataType} from '../../../lib/utils'
import AppContext from '../../../lib/contexts'
import { fetchers } from '../../../lib/apiFetchers'


const useStyles = makeStyles((theme) => {
  const borderColor = theme.palette.primary.light
  return {
    table: {
    textAlign: 'center',
    ...theme.mixins.gutters(),
    width: '100%',
    margin: 'auto'
  },
  td: {
    height: 200,
    position: 'relative',
  },
  borderBottom: {
    borderBottom: '1px solid',
    borderBottomColor: borderColor,
  },
  borderRight: {
    borderRight: '3px solid',
    borderRightColor: borderColor
  },
  arrow: {
    width: 0,
    height: 0,
    borderLeft: '8px solid transparent',
    borderRight: '8px solid transparent',
    borderTop: '10px solid',
    borderTopColor: borderColor,
    position: 'absolute',
    bottom: -2,
    right: -9,
  },
  circle: {
    width: 7,
    height: 7,
    backgroundColor: theme.palette.primary.dark,
    borderRadius: '50%',
    position: 'absolute',
    right: -5,
  }
}})

const makeTDDepth = (uiSoil, classes) => {

  const depths = uiSoil.map((us, i) => {
    return (<div key={us.d} style={{ height: us.h }}>
      <Typography variant="body1" children={us.d} />
      <div className={i === uiSoil.length - 1 ? classes.arrow : classes.circle} />
    </div>)
  })

  return depths
}

const makeTDTemp = (uiSoil) => {
  const temps = uiSoil.map((us, i) => {
    return (<div key={us.d} style={{ height: us.h }}>
      <Number number={us.v === null ? 'N/A' : us.v} />
    </div>)
  })

  return temps
}

const makeUiSoil = (agri) => {
  let uiSoil = [
    { h: '10%', d: '5cm', v: null },  // t5
    { h: '10%', d: '10cm', v: null }, // t10
    { h: '15%', d: '20cm', v: null }, // t20
    { h: '25%', d: '50cm', v: null }, // t50
    { h: '40%', d: '1m', v: null }, // t1m
  ]

  if (!agri) return uiSoil

  soilDataType.map((t, i) => uiSoil[i].v = (t in agri) ? agri[t] : null)

  return uiSoil
}

export default function Soil() {
  const classes = useStyles()
  const appContext = useContext(AppContext)

  const token = appContext.token
  const geo = appContext.geo

  const { data: stationSoil } = useSWR([geo, 31, token], fetchers.bomstn)
  const { data: agriSoil } = useSWR(() => [geo, stationSoil.data.station.id, token], fetchers.bomagri)

  if (!agriSoil || !agriSoil.data) {
    return <></>
  }

  let uiSoil = makeUiSoil(agriSoil.data.agri.soilTemp)

  return <>
    <table cellSpacing={0} className={classes.table}>
      <thead>
        <tr>
          <th className={clsx(classes.borderRight, classes.borderBottom)}>
            <Typography variant="body1" children='Depth' />
          </th>
          <th className={classes.borderBottom}>
            <Typography variant="body1" children='Temperature(Â°C)' />
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td className={clsx(classes.td, classes.borderRight)}>

            {makeTDDepth(uiSoil, classes)}
          </td>
          <td className={classes.td}>
            {makeTDTemp(uiSoil)}
          </td>
        </tr>
      </tbody>
    </table>
  </>
}