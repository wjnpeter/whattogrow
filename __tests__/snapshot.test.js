
import React from 'react';
import renderer from 'react-test-renderer';

import Home from '../pages/index'

import Forecast from '../components/search/Forecast'
import Moon from '../components/search/Moon'
import LocZone from '../components/search/loc/LocZone'
import Historical from '../components/search/loc/Historical'
import Agriculture from '../components/search/loc/Agriculture'
import Soil from '../components/search/loc/Soil'

test('Home page', () => {
  const component = renderer.create(<Home />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Forecast Popover', () => {
  const component = renderer.create(<Forecast />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Moon Popover', () => {
  const component = renderer.create(<Moon />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Location - LocZone', () => {
  const component = renderer.create(<LocZone />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Location - Historical', () => {
  const component = renderer.create(<Historical />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Location - Agriculture', () => {
  const component = renderer.create(<Agriculture />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('Location - Soil', () => {
  const component = renderer.create(<Soil />)
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
