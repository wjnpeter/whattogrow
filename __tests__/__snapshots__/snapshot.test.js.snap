// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Popover 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;

exports[`Home page 1`] = `
Array [
  <div
    className="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
  >
    <video
      autoPlay={true}
      className="makeStyles-background-3"
      muted={true}
      poster="/images/2461326-poster.jpg"
    >
      <source
        src="/images/2461326.mp4"
        type="video/mp4"
      />
    </video>
    <div
      className="makeStyles-overlay-5"
    />
    <button
      className="MuiButtonBase-root makeStyles-brand-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        height="32"
        src="/images/icons/logo.svg"
      />
    </button>
    <div
      className="MuiContainer-root makeStyles-content-6 MuiContainer-maxWidthSm"
    >
      <h1
        className="makeStyles-heading-2"
      >
        Search a selection of plants suit my location, climate and timing.
      </h1>
      <div
        aria-expanded={false}
        aria-owns={null}
        className="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
      >
        <div
          className="MuiBox-root MuiBox-root-53"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
          >
            <div
              className="MuiGrid-root makeStyles-gridLeft-21 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root makeStyles-textFieldInput-23 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-activedescendant={null}
                    aria-autocomplete="both"
                    aria-controls={null}
                    aria-invalid={false}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    placeholder="City | Town | Address"
                    required={false}
                    spellCheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    className="MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Clear"
                      className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      title="Clear"
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-213 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-214"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-gridRight-22 MuiGrid-item"
              onClick={[Function]}
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root makeStyles-gridRightIcon-19"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="MuiContainer-root makeStyles-footer-7 MuiContainer-maxWidthLg"
  >
    <span
      className="MuiTypography-root MuiTypography-caption"
    >
      May・Autumn・2020
    </span>
  </div>,
]
`;

exports[`Location - Agriculture 1`] = `null`;

exports[`Location - Historical 1`] = `
Array [
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>,
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>,
]
`;

exports[`Location - LocZone 1`] = `null`;

exports[`Location - Soil 1`] = `
<table
  cellSpacing={0}
  className="makeStyles-table-274"
>
  <thead>
    <tr>
      <th
        className="makeStyles-borderRight-277 makeStyles-borderBottom-276"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Depth
        </p>
      </th>
      <th
        className="makeStyles-borderBottom-276"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Temperature(°C)
        </p>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="makeStyles-td-275 makeStyles-borderRight-277"
      >
        <div
          style={
            Object {
              "height": "10%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            5cm
          </p>
          <div
            className="makeStyles-circle-279"
          />
        </div>
        <div
          style={
            Object {
              "height": "10%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            10cm
          </p>
          <div
            className="makeStyles-circle-279"
          />
        </div>
        <div
          style={
            Object {
              "height": "15%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            20cm
          </p>
          <div
            className="makeStyles-circle-279"
          />
        </div>
        <div
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            50cm
          </p>
          <div
            className="makeStyles-circle-279"
          />
        </div>
        <div
          style={
            Object {
              "height": "40%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            1m
          </p>
          <div
            className="makeStyles-arrow-278"
          />
        </div>
      </td>
      <td
        className="makeStyles-td-275"
      >
        <div
          style={
            Object {
              "height": "10%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            No data detected
          </p>
        </div>
        <div
          style={
            Object {
              "height": "10%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            No data detected
          </p>
        </div>
        <div
          style={
            Object {
              "height": "15%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            No data detected
          </p>
        </div>
        <div
          style={
            Object {
              "height": "25%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            No data detected
          </p>
        </div>
        <div
          style={
            Object {
              "height": "40%",
            }
          }
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            No data detected
          </p>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Moon Popover 1`] = `
<div
  className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
  role="progressbar"
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
>
  <svg
    className="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx={44}
      cy={44}
      fill="none"
      r={20.2}
      strokeWidth={3.6}
      style={Object {}}
    />
  </svg>
</div>
`;
